be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/MessageHandler.scala

```diff
@@ -58,4 +59,33 @@ trait MessageHandler extends AskPatternConstants {
       case _ => Future(Right(PipelineError(ErrorCodes.SERVER_ERROR.id, s"dbAskWritePropagate failed : could not write & propagate message $m", rpcRequest.id)))
     }
   }
+
+  /**
+   * Helper function for broadcasting messages to subscribers
+   *
+   * @param rpcMessage       : message for which we want to generate the broadcast
+   * @param channel          : the Channel in which we read the data
+   * @param broadcastData    : the message data we broadcast converted to Base64Data
+   * @param broadcastChannel : the Channel in which we broadcast
+   */
+  def dbBroadcast(rpcMessage: JsonRpcRequest, channel: Channel, broadcastData: Base64Data, broadcastChannel: Channel): GraphMessage = {
```

Comment:
I believe (feel free to correct me if I'm wrong) that we could simply use the broadcastChannel as an argument to get the LaoData with the ReadLaoData, since they are in the same LAO. This would remove the need for the channel argument

Reply:
True for now considering the changes for the SocialMedia and Lao, but for my other PR concerning the message KeyElection, we need to read in the laoChannel and the broadcast in the ElectionChannel. I preferred to do the change directly here eventhough it is not needed here.